lib/models/chat_mode.dart

```diff
@@ -0,0 +1,21 @@
+class ChatMode {
```

Comment:
Make this an enum: https://dart.dev/guides/language/language-tour#enumerated-types